client/blocks/upe/upe-deferred-intent-creation/payment-processor.js

```diff
@@ -194,6 +194,14 @@ const PaymentProcessor = ( {
 						};
 					}
 
+					const hasValidationError = document.querySelector(
```

Comment:
In block checkout, the validation check happens on the frontend. Checking for `.wc-block-components-validation-error` class gives us the result.